src/mode.js

```diff
@@ -13,16 +13,25 @@ const chromaticNotes = ['c', 'c#', 'd', 'd#', 'e', 'f', 'f#', 'g', 'g#', 'a', 'a
  * @return {[type]}                        Returns the mode as an array, e.g. ['c3', 'd3', 'e3', 'f3', 'g3', 'a3', 'b3', 'c4']
  */
 const mode = (root, mode, octave, addRootFromNextOctave) => {
-	root = root || 'c';
-	mode = mode || 'ionian';
-	octave = octave ? Number(octave) : 3;
-	addRootFromNextOctave = addRootFromNextOctave === false ? false : true;
+	if (root.match(/^[A-Za-z]\|[A-Za-z]+/)) {
+		var args = root.split('|');
+		root = args[0];
+		mode = args[1];
+		octave = args[2];
```

Comment:
Not sure how you got the actual file to be shown in your comment, but on line 32:
`octave = octave ? Number(octave) : 4;`

It is already defaulted to 4.